app/src/main/java/com/example/mymovie/MainActivity.java

```diff
@@ -1,169 +1,76 @@
+                .setDrawerLayout(drawer)
+                .build();
+        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
+        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration);
+        NavigationUI.setupWithNavController(navigationView, navController);
 
-    //리뷰초기화
-    public void setReviewData(){
-        adapter.addItem(new ReviewitemData(R.drawable.user1, 3.5f, "yeon**", "5분전", "적당히 재밌다. 오랜만에 잠 안오는 영화 봤네요.", "3"));
-        adapter.addItem(new ReviewitemData(R.drawable.user1, 4.7f, "kim**", "5분전", "적당히 재밌다.", "5"));
-        adapter.addItem(new ReviewitemData(R.drawable.user1, 4.7f, "cho**", "5분전", "적당히 재밌다.", "5"));
-        adapter.notifyDataSetChanged();
     }
-    //작성하기로 넘어가기
-    private void showCommendWriteActivity(){
-        Intent intent = new Intent(getApplicationContext(), CommendWriteActivity.class);
-        startActivityForResult(intent, WRITE_REQ_CODE);
-    }
-    //모두보기로 넘어가기
-    private void showCommendShowActivity(){
-        items = adapter.getItems();
-        Intent intent = new Intent(getApplicationContext(), CommendShowActivity.class);
-        intent.putParcelableArrayListExtra(ITEMS, items);
-        startActivityForResult(intent,SHOW_REQ_CODE);
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
+        return NavigationUI.navigateUp(navController, mAppBarConfiguration) || super.onSupportNavigateUp();
     }
 
     @Override
-    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent intent) {
-        super.onActivityResult(requestCode, resultCode, intent);
-
-        //작성하기에서 돌아왔을 때
-        if(requestCode==WRITE_REQ_CODE){
-            if(resultCode == Activity.RESULT_OK){
-                String commendText = intent.getStringExtra(COMMEND);
-                float commendRate = intent.getFloatExtra(RATE, 0.0f);
-                adapter.addItem(new ReviewitemData(R.drawable.user1, commendRate, "hi**", "10분전",commendText, "1"));
-                adapter.notifyDataSetChanged();
-            }
-        }
-        //모두보기에서 돌아왔을 때
-        else if(requestCode==SHOW_REQ_CODE){
-            items = intent.getParcelableArrayListExtra(ITEMS);
-            adapter.setItems(items);
-            adapter.notifyDataSetChanged();
+    public void onBackPressed() {
+        DrawerLayout drawer = findViewById(R.id.drawer_layout);
+        if (drawer.isDrawerOpen(GravityCompat.START)) {
+            drawer.closeDrawer(GravityCompat.START);
+        } else {
+            super.onBackPressed();
         }
     }
 
+    public void changeFragment(int resId, String name, String rate){
+        DetailFragment fragment = new DetailFragment();
+        Bundle bundle = new Bundle();
+        bundle.putInt("resId",resId);
+        bundle.putString("name",name);
+        bundle.putString("rate",rate);
```

Comment:
```suggestion
        DetailFragment fragment = new DetailFragment();
        Bundle bundle = new Bundle();
        bundle.putInt("resId",resId);
        bundle.putString("name",name);
        bundle.putString("rate",rate);
```

MovieData 의 모든 변수를 넘기는데 이렇게 하나하나 나눈 이유가 있을까요?
그냥 MovieData 자체를 넘기면 이렇게 중간 중간 하드 코딩이 들어가 휴먼 에러를 발생 시킬 수 있을 것 같습니다.